src/mscorlib/src/System/Reflection/Assembly.CoreCLR.cs

```diff
@@ -101,14 +101,16 @@ public static Assembly Load(AssemblyName assemblyRef)
         {
             Contract.Ensures(Contract.Result<Assembly>() != null);
             Contract.Ensures(!Contract.Result<Assembly>().ReflectionOnly);
-
-            if (assemblyRef != null && assemblyRef.CodeBase != null)
+            
+            AssemblyName modifiedAssemblyRef = null;
+            if (assemblyRef != null)
             {
-                throw new NotSupportedException(SR.NotSupported_AssemblyLoadCodeBase);
+                modifiedAssemblyRef = (AssemblyName)assemblyRef.Clone();
```

Comment:
Clone is not a particularly cheap operation. It would be nice to do this only if `assemblyRef.CodeBase` non-null.